### PR TITLE
desk-exec: init at 1.0.2

### PR DESCRIPTION
**Pull request by AxerTheAxe**
Hello, I would like to package my program `desk-exec`. 

`desk-exec` is a robust CLI-based application launcher for Linux. It provides an efficient way to search and execute desktop entries directly from the command line, with support for filtering by program names, substrings, mimetypes, and more. Designed for terminal-focused workflows, it follows the XDG Base Directory Specification, ensuring compatibility across Linux distributions.

I created `desk-exec` because I couldn't find a CLI tool that met my needs for launching applications based on desktop entries. Existing tools are either GUI-focused or lack the fine-grained filtering capabilities required for power users. This utility fills that gap while remaining lightweight and fast.

### Unique Features:
- Seamless integration with the XDG Base Directory Specification.
- Advanced filtering options (name, substring, mimetype, etc.).
- Minimal dependencies, making it suitable for a variety of environments.

I believe `desk-exec` is general-purpose and lightweight enough to be included in nixpkgs.

This is my first PR so sorry if I get anything wrong. Thanks!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AxerTheAxe** [comment]
> otherwise, LGTM. Welcome to nixpkgs.

Thanks! I removed `with lib;` and added `mainProgram`. I do not have a change log at the moment but I will be sure to add it here if I ever get around to creating one.

**Comment by AxerTheAxe** [comment]
> LGTM

Thanks again!

File: pkgs/by-name/de/desk-exec/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    hash = "sha256-bJLdd07IAf+ba++vtS0iSmeQSGygwSVNry2bHTDAgjE=";
+  };
+
+  cargoHash = "sha256-pK1WDtGkpPZgFwEm59TqoH7EkKPQivNuvsiOG0dbum4=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion target/${stdenv.targetPlatform.config}/release/dist/desk-exec.{bash,fish}
+    installManPage target/${stdenv.targetPlatform.config}/release/dist/desk-exec.1
```
Comment (Aleksanaa) [inline pkgs/by-name/de/desk-exec/package.nix:26]:
```suggestion
    installShellCompletion target/${stdenv.hostPlatform.config}/release/dist/desk-exec.{bash,fish}
    installManPage target/${stdenv.hostPlatform.config}/release/dist/desk-exec.1
```

See https://nixos.org/manual/nixpkgs/unstable/#ssec-cross-platform-parameters.

Or better:

```suggestion
    pushd target/${stdenv.hostPlatform.config}/release/dist
      installShellCompletion desk-exec.{bash,fish}
      installManPage desk-exec.1
    popd
```

File: pkgs/by-name/de/desk-exec/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  stdenv,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "desk-exec";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "AxerTheAxe";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/de/desk-exec/package.nix:15]:
```suggestion
    repo = "desk-exec";
```

File: pkgs/by-name/de/desk-exec/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    hash = "sha256-bJLdd07IAf+ba++vtS0iSmeQSGygwSVNry2bHTDAgjE=";
+  };
+
+  cargoHash = "sha256-pK1WDtGkpPZgFwEm59TqoH7EkKPQivNuvsiOG0dbum4=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    pushd target/${stdenv.hostPlatform.config}/release/dist
+      installShellCompletion desk-exec.{bash,fish}
```
[the remark this answers is not in the document]
Comment (AxerTheAxe) [reply pkgs/by-name/de/desk-exec/package.nix:26]: > Did you forget to push?

[I don't think so?](https://github.com/NixOS/nixpkgs/compare/35a69d80341335316ae4dfd81d046c1b9a556150..14201d3a9d46f2b4362fe5ad3b08053830c25fd7) `installShellCompletion _desk-exec` installs the completion for ZSH. `_desk-exec` is what ZSH expects and is what desk-exec produces on build. `installShellCompletion` has the ability to convert a `desk-exec.zsh` into `_desk-exec` automatically but changing the completions format would require me to update the completions everywhere else I have desk-exec packaged so I did not find changing it to be worth while. However, if `desk-exec.zsh` is preferred in Nixpkgs I would be okay with updating the completions :)

**Comment by Aleksanaa** [comment]
@ofborg build desk-exec

**Pull request merged**